### PR TITLE
[APIDX-796] Added notifications on build success and failure

### PR DESCRIPTION
**Pull request by arturrudiuk-criteo**

### CODE REVIEW

File: .github/workflows/generate_java_sources.yml [hunk 1]
```diff
@@ -31,4 +31,15 @@ jobs:
...
         with:
           name: generated-sources-java
-          path: ./generated-sources/java
\ No newline at end of file
+          path: ./generated-sources/java
+      
+      - name: Send success notification
+        run: |
+          chmod +x ./scripts/send-sdk-build-notification.sh
+          ./scripts/send-sdk-build-notification.sh generate_java_sources.yml "Java build succeeded" green_check_mark ${{ secrets.SLACK_WEBHOOK }}
```
Comment (PaulMathon) [inline .github/workflows/generate_java_sources.yml:39]: you can define environment variables and use them in shell scripts, maybe this is more adapted than passing a secret as a command-line argument?
https://github.community/t/using-github-action-environment-variables-in-shell-script/18330

File: .github/workflows/generate_python_sources.yml [hunk 2]
```diff
@@ -31,4 +31,15 @@ jobs:
...
         with:
           name: generated-sources-python
-          path: ./generated-sources/python
\ No newline at end of file
+          path: ./generated-sources/python
+
+      - name: Send success notification
+        run: |
+          chmod +x ./scripts/send-sdk-build-notification.sh
+          ./scripts/send-sdk-build-notification.sh generate_python_sources.yml "Python build succeeded" green_check_mark ${{ secrets.SLACK_WEBHOOK }}
```
Comment (PaulMathon) [inline .github/workflows/generate_python_sources.yml:39]: same about the environment variable

File: .github/workflows/generate_php_sources.yml [hunk 3]
```diff
@@ -31,4 +31,15 @@ jobs:
...
         with:
           name: generated-sources-php
-          path: ./generated-sources/php
\ No newline at end of file
+          path: ./generated-sources/php
+
+      - name: Send success notification
+        run: |
+          chmod +x ./scripts/send-sdk-build-notification.sh
+          ./scripts/send-sdk-build-notification.sh generate_php_sources.yml "PHP build succeeded" green_check_mark ${{ secrets.SLACK_WEBHOOK }}
```
Comment (PaulMathon) [inline .github/workflows/generate_php_sources.yml:39]: same about the environment variable

File: scripts/send-sdk-build-notification.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+RUN_ID="$(curl https://api.github.com/repos/criteo/criteo-api-sdk-generator/actions/workflows/$1/runs \
+            | jq -r '.workflow_runs[0].id | tostring')"
+
+result=$(curl -X POST --data-urlencode \
+"payload={ \
+    \"channel\": \"#criteo-api-sdk-generator\", \
+    \"username\": \"sdk-generation-bot\", \
+    \"text\": \"$2.\n<https://github.com/criteo/criteo-api-sdk-generator/actions/runs/${RUN_ID}|Link to build>\", \
+    \"icon_emoji\": \":$3:\"}" \
+    $4 )
```
Comment (PaulMathon) [inline scripts/send-sdk-build-notification.sh:13]: Could be SLACK_WEBHOOK if we pass it as env variable (clearer from my pov)

File: scripts/send-sdk-build-notification.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+RUN_ID="$(curl https://api.github.com/repos/criteo/criteo-api-sdk-generator/actions/workflows/$1/runs \
+            | jq -r '.workflow_runs[0].id | tostring')"
+
+result=$(curl -X POST --data-urlencode \
+"payload={ \
+    \"channel\": \"#criteo-api-sdk-generator\", \
+    \"username\": \"sdk-generation-bot\", \
+    \"text\": \"$2.\n<https://github.com/criteo/criteo-api-sdk-generator/actions/runs/${RUN_ID}|Link to build>\", \
+    \"icon_emoji\": \":$3:\"}" \
```
Comment (PaulMathon) [inline scripts/send-sdk-build-notification.sh:12]: maybe we could set a variable called for example `emoji=$3` and use it here?

File: .github/workflows/generate_python_sources.yml [hunk 6]
```diff
@@ -31,4 +31,15 @@ jobs:
         uses: actions/upload-artifact@v2
         with:
           name: generated-sources-python
-          path: ./generated-sources/python
\ No newline at end of file
+          path: ./generated-sources/python
+
+      - name: Send success notification
```
Comment (PaulMathon) [inline .github/workflows/generate_python_sources.yml:36]: this action will not be run in case of failure right?

**Review by PaulMathon — APPROVED** [review]

**Review by ouvreboite — APPROVED** [review]

**Pull request merged**